common/tools/patch_tool_test.sh

```diff
@@ -26,7 +40,12 @@ readonly MY_EXPECT_FILE
   echo "Expecting 1 positional argument, verible-patch-tool path."
   exit 1
 }
-patch_tool="$1"
+
+# On Windows, runfiles are not symlinked (unless --enable_runfiles is active,
+# which causes other issues on bazel 3.7.0), so use rlocation to find the path
+# to the executable. On Unix, you could simply use $1, but using rlocation
+# works too
```

Comment:
Ok to understand the `$(location :verible-patch-tool)` we use in the BUILD rule only outputs the relative path from the start of toplevel `bazel-bin`, but since that is in the Windows case not in the PATH, we need to use `rlocation`  to resolve ?

(Does it make a difference with Bazel 4.0 ?)

Reply:
I'm not a bazel expert, but I think bazel creates a folder to run the script (kinda like a sandbox) and make `deps` files available using symlinks. So the `$(location :verible-patch-tool)` is the location of the symlinked script from the sandbox folder.

Except on Windows it doesn't create the symlink, but instead create a `MANIFEST` file (also available on Unix) that has a mapping of all available runfiles to its fullpath. `rlocation` is a convenience function that look inside the `MANIFEST` file for a relative runfile path and returns the fullpath.

And no, bazel 4.0 doesn't make a difference.